lib/android/src/main/java/com/airbnb/android/react/navigation/ScreenCoordinatorLayout.java

```diff
@@ -66,38 +66,54 @@ public void setFragmentManager(FragmentManager fragmentManager) {
     this.fragmentManager = fragmentManager;
   }
 
-  @Override
-  public void addView(View child, int index, ViewGroup.LayoutParams params) {
-    int backStackEntryCount = fragmentManager.getBackStackEntryCount();
-    reverseDrawing = backStackEntryCount > previousBackStackEntryCount;
-    previousBackStackEntryCount = backStackEntryCount;
-    super.addView(child, index, params);
+  public void willDetachCurrentScreen() {
+    isDetachingCurrentScreen = true;
   }
 
   @Override
-  public void removeView(View view) {
+  public void removeView(final View view) {
+    if (isDetachingCurrentScreen) {
+      isDetachingCurrentScreen = false;
+      reverseLastTwoChildren = true;
+    }
+
     super.removeView(view);
   }
 
   @Override
   protected void dispatchDraw(Canvas canvas) {
     super.dispatchDraw(canvas);
 
-    while (!drawingOps.isEmpty()) {
-      DrawingOp op = drawingOps.remove(drawingOps.size() - 1);
+    if (drawingOps.size() < previousChildrenCount) {
+      reverseLastTwoChildren = false;
+    }
+    previousChildrenCount = drawingOps.size();
+
+    while (drawingOps.size() > 2) {
```

Comment:
1) Maybe we can make a `drawAndRelease` helper
2) Can this loop just be in the else case and loop until !empty? Seems like all it's doing is finishing this loop. Then make the if case include this loop plus what's currently there.